build/signDSCModules.sh

```diff
@@ -64,7 +69,7 @@ DscModuleIntermediateDir="$IntermediateDir/merging_dsc_modules"
 DscModuleTargetDir="$TargetDir/dsc_signed"
 X64Dir="$BUILD_BASEDIR/Linux_ULINUX_1.0_x64_64_Release/dsc"
 X86Dir="$BUILD_BASEDIR/Linux_ULINUX_1.0_x86_32_Release/dsc"
-TestSigningDir=$1
+TestSigningDir=$2
```

Comment:
Line 23 (now line 28 due to your new comments) is a weird way to handle parameters, and won't really do what you want.

If you want the notion of two separate optional parameters (which is probably correct here), then you should allow them independently of one another. That is:

- No parameters need be specified if you don't need to override,
- Only the build share can be specified,
- Only the test signing dir can be specified,
- Both the build share and test signing dir can be specified.

The way the code is currently structured, this isn't the case today. Please fix.